A36-xds-for-servers.md

```diff
@@ -0,0 +1,332 @@
+A36: xDS-Enabled Servers
+------------------------
+* Author(s): [Eric Anderson](https://github.com/ejona86), [Doug
+  Fawley](https://github.com/dfawley), [Mark Roth](https://github.com/markdroth)
+* Approver: markdroth
+* Status: Draft {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: <language, ...>
+* Last updated: 2020-11-19
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+Bring xDS-based configuration to Servers. xDS has many features and this gRFC
+will not provide support for any of them in particular. However, it will provide
+the central plumbing and APIs to add support for specific features and have them
+work without additional user code changes.
+
+There's multiple pieces involved:
+
+*  The API surface exposed to users to enable the new system.
+*  The gRPC-internal plumbing to allow injecting behavior into servers, and
+   changing that behavior on-the-fly.
+*  The specific xDS protocol behavior gRPC will use to retrieve configuration.
+
+## Background
+
+Since [gRFC A27: xDS-Based Global Load Balancing][A27], Channels have had
+growing support for understanding xDS configuration and configuring themselves.
+However, Servers are also important to connect to xDS in order to power TLS
+communication, authorization, fault injection, rate limiting, and other
+miscellaneous features.
+
+On client-side, users are able to opt-in to the behavior via the `xds:` scheme
+which enables the xDS Name Resolver. Servers in gRPC do not use Name Resolution
+and certainly don't have things like Load Balancing APIs which have been
+leveraged heavily on the client-side design. While server-side will need to use
+a different approach, users should have a similar experience: they make a
+trivial code change to opt-in to xDS, and from that point on newly-added
+xDS-powered gRPC features can be used without code changes.
+
+### Related Proposals:
+
+ * [A27: xDS-Based Global Load Balancing][A27]
+ * [A29: xDS-Based Security for gRPC Clients and Servers][A29]
+
+[A27]: A27-xds-global-load-balancing.md
+[A29]: A29-xds-tls-security.md
+
+## Proposal
+
+Each language will create an "XdsServer" API that will wrap the normal "Server"
+API. This will allow it to see the listening ports, start event, stop event, and
+also allow it access to the server to change behavior (e.g., via an
+interceptor). Users will use the XdsServer API to construct the server instead
```

Comment:
I'm not sure this is true for Go, at least.  We don't have hooks for start/stop, and the current xds server API doesn't allow injecting interceptors or direct access to the underlying grpc server.